lib/crypto.js

```diff
@@ -772,7 +779,9 @@ Crypto.prototype.decryptEvent = function(event) {
     if (!AlgClass) {
         throw new algorithms.DecryptionError("Unable to decrypt " + content.algorithm);
     }
-    var alg = new AlgClass(this._deviceId, this, this._olmDevice);
+    var alg = new AlgClass({
+        olmDevice: this._olmDevice,
+    });
```

Comment:
The others weren't used, and were originally cargo-culted from EncryptionAlgorithm. I thought I would leave them out until I found a use for them, rather than including them in case they were useful.